src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -537,8 +539,11 @@ void RealmCoordinator::init_external_helpers()
 #if REALM_ENABLE_SYNC
     // We may have reused an existing sync session that outlived its original
     // RealmCoordinator. If not, we need to create a new one now.
-    if (m_config.sync_config && !m_sync_session)
-        m_sync_session = m_config.sync_config->user->sync_manager()->get_session(m_db, m_config);
+    if (m_config.sync_config && !m_sync_session) {
+        if (auto app = m_config.sync_config->user->backing_store()->app().lock()) {
```

Comment:
If this is somehow null we'll just silently not create a session which probably isn't correct?